### PR TITLE
Full drunc image for controller UI development

### PR DESCRIPTION
**Pull request by cc-a**
# Description

Expands the docker image used by the `drunc` service to provide a fully functioned version of drunc capable of booting realistic test sessions. This is only required for development of the controller UI so we should decide whether we're doing that in this repository or as a separate app before merging.

Fixes # (issue)

## Type of change

- [ ] Documentation (non-breaking change that adds or improves the documentation)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Optimization (non-breaking, back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (whatever its nature)

## Key checklist

- [ ] All tests pass (eg. `python -m pytest`)
- [ ] The documentation builds and looks OK (eg. `python -m sphinx -b html docs docs/build`)
- [ ] Pre-commit hooks run successfully (eg. `pre-commit run --all-files`)

## Further checks

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added or an issue has been opened to tackle that in the future. (Indicate issue here: # (issue))

### CODE REVIEW

File: drunc_docker_service/Dockerfile [hunk 1]
```diff
@@ -1,21 +1,37 @@
...
+
+# for frozen release
+RUN source /cvmfs/dunedaq.opensciencegrid.org/setup_dunedaq.sh && \
+        setup_dbt latest_v5 && \
+        dbt-create fddaq-v5.1.0-a9 fddaq-v5.1.0-a9 && \
+        cd fddaq-v5.1.0-a9 && \
+        source env.sh && \
+        dbt-build && \
+        dbt-workarea-env && \
+        pip install git+https://github.com/DUNE-DAQ/drunc.git@v0.10.2
```
[the remark this answers is not in the document]
Comment (cc-a) [reply drunc_docker_service/Dockerfile:18]: But that would require me to understand what this is doing. Which I don't 🙃.

**Comment by cc-a** [comment]
> Is there any way that we could give devs the choice whether to use the full/lite images? E.g. by providing arguments to a helper script?

Definitely could. Worth doing if we decide to keep everything in one repo.

**Review by jamesturner246 — APPROVED** [review]
Makes sense at the code level. I'll revise my comment if all breaks when I run after downloading.

**Comment by jamesturner246** [comment]
Looks good now :+1:

**Comment by TinyMarsh** [comment]
If someone could double check the latest merge commit that would be great. I think it's fine but may have missed something when the conflicts got a bit weird specifically in the README.

If all is okay then I can merge this in to #147

**Comment by cc-a** [comment]
LGTM @TinyMarsh

**Comment by cc-a** [comment]
OK, with the help of @TinyMarsh I believe this is now ready for full review. I'll re-request from poeple who've already done one and pinging @AdrianDAlessandro for visibility.

File: README.md [hunk 2]
```diff
@@ -11,32 +11,85 @@ This repo defines the web interface for various drunc tools. Including:
...
 
-It can take a few moment for the services to boot but the application should then be
-available in the browser at <http://localhost:8000>.  Authentication is required to work
-with the application so you need to create a user account to work with:
+It can take a few moments for the services to boot but the application should then be
+available in the browser at <http://localhost:8000>.
+
+__Note that if you have the Process Manager UI already open in a browser you may see
+backtraces from unsuccessful requests generated by the UI. These should stop once drunc
+has finished starting up.__
```
Comment (alexdewar) [inline README.md:51]: Are the backtraces full stack traces? Will users be able to see it in production....? If so, is that what we want?

File: drunc_docker_service/boot_test_session.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+# setup dunedaq environment
+. /basedir/fddaq-v5.1.0-a9/env.sh
+
+# boot full test session
+drunc-process-manager-shell grpc://localhost:10054 boot test/config/test-session.data.xml test-session
```
Comment (alexdewar) [inline drunc_docker_service/boot_test_session.sh:7]: Minor nit:

```suggestion
exec drunc-process-manager-shell grpc://localhost:10054 boot test/config/test-session.data.xml test-session
```

**Review by alexdewar — APPROVED** [review]
I've made some small comments but otherwise LGTM :+1:

File: README.md (hunk 2 above)
Comment (cc-a) [reply README.md:51]: Ah, maybe I should be clearer here. It's only in the logs in the terminal.

Comment (alexdewar) [reply README.md:51]: Ohh I see. That's not such a big deal

**Comment by jamesturner246** [comment]
Having some weird behaviours:

```sh
docker compose --profile drunc up
```

ends in 

```
Error response from daemon: network 2892bd9abdea2fdd8cff6e493db78d55d98a11a8825874ccdc894ca1ee03344c not found
```

Also there is no indication that drunc or drunc-lite are even selected when running, or even if neither at all are chosen! Running without either seems to run fine. :shrug:

**Comment by alexdewar** [comment]
@jamesturner246 I don't see that error, but I am getting this one (Kafka related?):

```
kafka_consumer-1  | Listening for messages from Kafka.
kafka_consumer-1  | Traceback (most recent call last):
kafka_consumer-1  |   File "/usr/local/lib/python3.11/site-packages/django/db/backends/utils.py", line 105, in _execute
kafka_consumer-1  |     return self.cursor.execute(sql, params)
kafka_consumer-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
kafka_consumer-1  |   File "/usr/local/lib/python3.11/site-packages/django/db/backends/sqlite3/base.py", line 354, in execute
kafka_consumer-1  |     return super().execute(query, params)
kafka_consumer-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

(etc. etc.)

I also tried just running `docker compose up` without setting `--profile` to see what would happen and it ran the full image. Maybe it defaults to whatever profile you selected last?

**Comment by cc-a** [comment]
Going to assume you closed this by accident @alexdewar.

**Comment by cc-a** [comment]
@alexdewar @jamesturner246 docker compose can sometimes get into a funny state if not all resources have been torn down previously and changes are made to the configuration. Could you both try a `docker compose down -v --remove-orphans` and take another pass.

The expected behaviour should be that:
- running with `--profile` should run the relavant version of the drunc service
- running without `--profile` should run all other services. In this case the app service will still start successfully and all views that don't involve communicating with the process manager will work. Opening the process manager UI will load the page but the table and message feed will not be populated (improvements to this planned in #150).

**Comment by alexdewar** [comment]
> Going to assume you closed this by accident @alexdewar.

Yes, sorry!

**Review by jamesturner246 — APPROVED** [review]
Aha, that was it. It all works fine then.

Just need to remember to spell the profile correctly, as there's no indication that the given profile name is wrong.

**Comment by cc-a** [comment]
With this merged you will need to update your local development workflow to either set the `COMPOSE_PROFILES` environment variable or use the `--profile` flag with `docker compose`.

**Pull request merged**